go-controller/pkg/informer/informer.go

```diff
@@ -161,6 +161,10 @@ func (e *eventHandler) Run(threadiness int, stopCh <-chan struct{}) error {
 
 // enqueue adds an item to the workqueue
 func (e *eventHandler) enqueue(obj interface{}) {
+	// ignore objects that are already set for deletion
```

Comment:
@aojea I'm not familiar with the details, but at this point have we already processed the deletion event in the handlers when this case is hit?

Reply:
This is from the log where I've observed this behavior
```
ovn-kubernetes/ovnkube-master.log-I0810 16:26:22.776583      48 namespace.go:234] Updating namespace: dns-9831
ovn-kubernetes/ovnkube-master.log:I0810 16:26:22.789874      48 namespace.go:320] Deleting namespace: dns-9831
```
both events are practically consecutive, and if IIRC the namespaces use finalizers for the graceful deletion of all the objects

The delete is received just 10ms later
https://github.com/ovn-org/ovn-kubernetes/blob/6f0241aa0b75884773ebd6ec5f9ee0c8d01805e2/go-controller/pkg/informer/informer.go#L88-L107

This may explain some race conditions